utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java

```diff
@@ -43,20 +49,52 @@ public static void main(String[] args) throws ConfigurationException, IOExceptio
     public void run() throws CodecException, IOException, ConfigurationException, TimeoutException
     {
         final String familyReference = getEnvironmentValue(FamilyTaskSubmitterConstants.FAMILY_REFERENCE); 
-        final ElasticQuery esQuery = new ElasticQuery(familyReference);
+        final DocumentWorkerDocument rootDocument = buildDocument(familyReference);
+        final String outputPartialReference = getEnvironmentValue(FamilyTaskSubmitterConstants.Message.OUTPUT_PARTIAL_REFERENCE);
+        final String projectId = getEnvironmentValue(FamilyTaskSubmitterConstants.Message.PROJECT_ID);
+        final long workflowId = Long.parseLong(getEnvironmentValue(FamilyTaskSubmitterConstants.Message.WORKFLOW_ID));
+        submitTask(rootDocument, outputPartialReference, projectId, workflowId);
+    }
+
+    /**
+     * Builds a family consisting of a root family document and its sub-documents based on provided family reference.
+     * @param familyReference Family Reference field value that represents the family and can be used to build document
+     *                        family.
+     * @return Constructed family of documents.
+     */
+    private static DocumentWorkerDocument buildDocument(String familyReference) {
+        final ElasticQuery esQuery;
+        try {
+            esQuery = new ElasticQuery(familyReference);
+        } catch (ConfigurationException e) {
+            throw new RuntimeException("Failure to configure Elasticsearch query with family reference: "+familyReference,
+                    e);
+        }
         final Map<String, Object> family = esQuery.createFamily(familyReference);
         final DocumentExtractor docBuilder = new DocumentExtractor(family);
-        final Map<String, Multimap<String, String>> documents = docBuilder.convert();
+        final Map<String, Multimap<String, String>> documentsMetadataMap = docBuilder.convert();
         final HierarchyBuilder hierarchyBuilder = new HierarchyBuilder();
-        final Document rootDocument = hierarchyBuilder.convert(documents, familyReference);
-        final Message message = new Message(rootDocument, getEnvironmentValue(FamilyTaskSubmitterConstants.OUTPUT_QUEUE_NAME));
-        final RabbitMessageDispatcher messageDispatcher = new RabbitMessageDispatcher();
-        messageDispatcher.configure(getEnvironmentValue(FamilyTaskSubmitterConstants.OUTPUT_QUEUE_NAME));
-        messageDispatcher.sendMessage(message.createTaskMessage());
+        return hierarchyBuilder.convert(documentsMetadataMap, familyReference);
     }
 
     private static String getEnvironmentValue(final String name)
     {
         return System.getProperty(name, System.getenv(name) != null ? System.getenv(name) : "");
     }
+
+    private static void submitTask(DocumentWorkerDocument documentToSend,
```

Comment:
The other parameters are `final` but this one is not.

Reply:
Fixed.